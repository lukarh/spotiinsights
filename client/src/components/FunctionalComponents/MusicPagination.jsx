import Pagination from "@mui/material/Pagination";

const MusicPagination = ({ recentPlaylist, itemsPerPage, currentPage, handlePageChange }) => {
    return (
        <div className="flex-center padding-1-rem">
            <Pagination 
                count={Math.ceil(recentPlaylist.length / itemsPerPage)} 
                color="success" 
                showFirstButton 
                showLastButton 
                page={currentPage}
                onChange={handlePageChange}
                sx={{ 
                    button: { color: '#ffffff', fontFamily: "DM Sans" },
                }}
            />
        </div>
    )
}

export default MusicPagination;