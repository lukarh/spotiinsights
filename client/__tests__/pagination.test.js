import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { act, render, cleanup, fireEvent } from "@testing-library/react";
import mockRecentPlaylistData from '../__testsData__/mockRecentPlaylistData';
import MusicPagination from '../src/components/FunctionalComponents/MusicPagination';

describe('Music Pagination Component', () => {

    beforeEach(() => {
        jest.clearAllMocks()
    })

    afterEach(() => {
        cleanup()
    })

    const itemsPerPage = 7
    const mockHandlePageChange = jest.fn()

    test('should render the pagination component regardless of no input array data and itemsPerPage (7) set statically', async () => {
        const mockData = []
        const currentPage = 0

        const { getByRole } = render(
            <MusicPagination recentPlaylist={mockData} itemsPerPage={itemsPerPage} currentPage={currentPage} onChange={mockHandlePageChange} />
        )
        
        const firstPageButton = getByRole('button', { name: 'Go to first page' })
        const previousPageButton = getByRole('button', { name: 'Go to previous page' })
        const nextPageButton = getByRole('button', { name: 'Go to next page' })
        const lastPageButton = getByRole('button', { name: 'Go to last page' })

        expect(firstPageButton).toBeInTheDocument()
        expect(previousPageButton).toBeInTheDocument()
        expect(nextPageButton).toBeInTheDocument()
        expect(lastPageButton).toBeInTheDocument()
        expect(mockHandlePageChange).not.toHaveBeenCalled()
    })

    test('should render the pagination component with correct no. of pages for a given input array (length 50) and itemsPerPage (7) set statically', async () => {
        const currentPage = 1

        const { getByRole } = render(
            <MusicPagination recentPlaylist={mockRecentPlaylistData} itemsPerPage={itemsPerPage} currentPage={currentPage} onChange={mockHandlePageChange} />
        )

        const totalPages = Math.ceil(mockRecentPlaylistData.length / itemsPerPage)
        const totalPagesButton = getByRole('button', { name: `Go to page ${totalPages}` })

        await act(async () => {
            fireEvent.click(totalPagesButton)
        })

        expect(totalPagesButton).toBeInTheDocument()
    })

    test('should render the pagination component with correct no. of pages for a given input array (length 15) and itemsPerPage (7) set statically', async () => {
        const currentPage = 1
        const mockData = mockRecentPlaylistData.slice(0,15)

        const { getByRole } = render(
            <MusicPagination recentPlaylist={mockData} itemsPerPage={itemsPerPage} currentPage={currentPage} onChange={mockHandlePageChange} />
        )

        const totalPages = Math.ceil(mockData.length / itemsPerPage)
        const totalPagesButton = getByRole('button', { name: `Go to page ${totalPages}` })

        await act(async () => {
            fireEvent.click(totalPagesButton)
        })

        expect(totalPagesButton).toBeInTheDocument()
    })

    test('should match snapshot when given an input array (length 50)', () => {
        const currentPage = 3
        const tree = renderer.create(
            <MusicPagination recentPlaylist={mockRecentPlaylistData} itemsPerPage={itemsPerPage} currentPage={currentPage} onChange={mockHandlePageChange} />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should match snapshot when given an input array (length 15)', () => {
        const currentPage = 1
        const mockData = mockRecentPlaylistData.slice(0,15)
        const tree = renderer.create(
            <MusicPagination recentPlaylist={mockData} itemsPerPage={itemsPerPage} currentPage={currentPage} onChange={mockHandlePageChange} />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
