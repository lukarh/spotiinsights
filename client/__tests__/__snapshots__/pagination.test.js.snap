// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music Pagination Component should match snapshot when given an input array (length 15) 1`] = `
<div
  className="flex-center padding-1-rem"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root MuiPagination-text css-1dzdshj-MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to first page"
          className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess Mui-disabled MuiPaginationItem-firstLast css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="FirstPageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess Mui-disabled MuiPaginationItem-previousNext css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess Mui-selected MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          2
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          3
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-previousNext css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to last page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-firstLast css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="LastPageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Music Pagination Component should match snapshot when given an input array (length 50) 1`] = `
<div
  className="flex-center padding-1-rem"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root MuiPagination-text css-1dzdshj-MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to first page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-firstLast css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="FirstPageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-previousNext css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 1"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          1
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          2
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 3"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess Mui-selected MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          3
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          4
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          5
        </button>
      </li>
      <li>
        <div
          className="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
        >
          …
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 8"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-page css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          8
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-previousNext css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to last page"
          className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textSuccess MuiPaginationItem-firstLast css-gmafwt-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="LastPageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
